### DISCUSSION TITLE
[8.0.0-beta] Parsing error: Unexpected token =, on class fields

### DISCUSSION THREAD

**Discussion by trusktr** [eslint/eslint#15049] [Q&A] [answered]
It's 2021, so I don't think that this error should be happening on class properties out of the box. Is my config object wrong?

```json
{
  "env": {
    "browser": true,
    "es2021": true,
    "mocha": true
  },
  "globals": {
    "EventEmitter2": true,
    "THREE": true,
    "ROS3D": true,
    "ROSLIB": true,
    "requestAnimationFrame": true,
    "cancelAnimationFrame": true
  },
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "semi": ["warn", "always"],
    "curly": "error",
    "eqeqeq": "error",
    "wrap-iife": ["error", "any"],
    "no-use-before-define": "off",
    "new-cap": "error",
    "no-caller": "error",
    "dot-notation": "off",
    "no-undef": "error",
    "no-cond-assign": "off",
    "no-eq-null": "off",
    "no-proto": "off",
    "no-console": "off",
    "no-unused-vars": "off",
    "strict": "off",
    "quotes": ["error", "single"],
    "linebreak-style": "error"
  }
}
```

I've set `es2021`. What else needs to be set to enable common syntax supported by all browsers?

**Comment by trusktr** [comment, marked as answer]
This works, but I really think it shouldn't be required these days:

```sh
npm i -D babel-eslint
```
Then in eslintrc:

```json
{
  "parser": "babel-eslint"
}
```

**Reply by mdjermanovic** [reply]
Class fields are ES2022 syntax. You can use default parser with `ecmaVersion: 2022` and all ES2022 syntax listed in [v8.0.0-beta.2 release blog post](https://eslint.org/blog/2021/09/eslint-v8.0.0-beta.2-released) will be supported.

```json
{
    "parserOptions": {
        "ecmaVersion": 2022
    }
}
```

Years relate to ECMAScript specifications. `"es2021": true` environment enables all built-in globals defined in ECMAScript 2021, and all syntax defined in ECMAScript 2021 (technically, by setting [`ecmaVersion`](https://github.com/eslint/eslintrc/blob/e051f50fd1e843537531392b492173ad82f6abc1/conf/environments.js#L90) in parser options), so it doesn't enable class fields.

We're adding new environments only if there are new globals in the spec, so there's still no `es2022` environment, but you can enable syntax by setting `ecmaVersion: 2022` in parser options.
